src/LinuxTracing/PerfEvent.h

```diff
@@ -201,6 +203,8 @@ using SchedSwitchPerfEvent = TypedPerfEvent<SchedSwitchPerfEventData>;
 
 struct SchedWakeupPerfEventData {
   pid_t woken_tid;
+  pid_t woker_tid;
+  pid_t woker_pid;
```

Comment:
Can these fields be "empty" - for example with a pid/tid of `0`? Or are they always filled? If not I would add a comment about that. Same for Newtask above.

Reply:
I think they can only be zero if the tracepoint cannot find the tid and pid. 

We already discussed cases where we don't have the pid but are there cases where we don't have the tid?

Reply:
Not sure. You tell me. :smile: 

But if there is a chance that these are zero, then I would add a comment explaining that this field is zero when the information is not available.